### DISCUSSION TITLE
goreleaser release error

### DISCUSSION THREAD

**Discussion by Zhang21** [goreleaser/goreleaser#2786] [Q&A] [answered]
```
goreleaser release
   • releasing...
   • loading config file       file=.goreleaser.yaml
   • loading environment variables
   • getting and validating git state
   ⨯ release failed after 0.02s error=git doesn't contain any tags. Either add a tag or use --snapshot
```

**Comment by Zhang21** [comment]
```
git tag -l
v0.1
v0.2
v0.3
v0.4
v0.5

```

**Comment by Zhang21** [comment]
I can't find any answers on google.

**Comment by Zhang21** [comment]
```
ls dist/
artifacts.json  checksums.txt  config.yaml  goreleaser-demo_0.0.0-snapshot_Linux_x86_64.tar.gz  goreleaser-demo_0.0.0-snapshot_Windows_x86_64.zip  goreleaser-demo_linux_amd64  goreleaser-demo_windows_amd64

```

**Comment by Zhang21** [comment]
```
goreleaser -v
goreleaser version dev
goos: linux
goarch: amd64
module version: v1.2.1, checksum: h1:s+HmalEPBtSOEjqTOXftIznxYl/JvleW9UkRQ4WXvjg=

https://goreleaser.com
```

**Comment by caarlos0** [comment, marked as answer]
this is not a question, it does not ask anything nor give any relevant details of what's happening or where goreleaser is running.

the answers are not answers as well, merely comments on the initial issue.

The error says it all:

```
git doesn't contain any tags
```

if its a CI, I would guess you're running against a shallow clone on the main/master branch instead of a tag - which can be avoided by following our documentation. https://goreleaser.com/ci/ 

Next time, please, formulate your question properly, what are you trying to do, where, how, code snippets, etc. Also please refer to the [pinned post on this forum](https://github.com/goreleaser/goreleaser/discussions/2571).

**Reply by Zhang21** [reply]
OK.

**Reply by Zhang21** [reply]
OK.

**Comment by Zhang21** [comment]
The problem is the version of git，the answer is update git.

<br/>

I run debug use `goreleaser release --debug`，I find the problem and answer.

``` 
• running git               args=[-c log.showSignature=false tag --points-at HEAD --sort -version:refname]
      • git result                stderr=error: unknown option `sort'
usage: git tag [-a|-s|-u <key-id>] [-f] [-m <msg>|-F <file>] <tagname> [<head>]
   or: git tag -d <tagname>...
   or: git tag -l [-n[<num>]] [--contains <commit>] [--points-at <object>]
                [<pattern>...]
   or: git tag -v <tagname>...
```

It shows git no `--sort` option，the version of git。

The default version of git on centos7 is `1.18.3`，so I update my git to 2.31.1.

**Reply by gtirloni** [reply]
That was exactly my problem, thank you. Once I updated git on CenOS 7, everything worked.
